File: RA1/clase12/JAVA/Erizo.java
public class Erizo extends Animal{
    private int cantPuas;

    public Erizo(String nombre, String especie, String raza, int edad, double peso, int cantPuas){
        super(nombre, especie, raza, edad, peso);
        this.cantPuas = cantPuas;
    }

    public void seHaceBolita(){
        System.out.println(this.nombre + " se hace Bolita y te puede pinchar...");
    }

    //Metodo por sobrecarga
    public void sleep(){
        System.out.println(this.nombre + " es un erizo que duerme... zzz ");
    }

    @Override
    public void makeSound(){
        System.out.println(this.nombre + " hace quikquik...");
    }
}

File: RA1/clase12/PYTHON/Erizo.py
from Animal import Animal

class Erizo(Animal):

    def __init__(self, nombre, especie, raza, edad, peso, cant_puas):
        super().__init__(nombre, especie, raza, edad, peso)
        self.cant_puas = cant_puas

    def sleep(self):
        print(self.nombre + " es un erizo que esta duermiendo.. zzz")
    
    def make_sound(self):
        print(self.nombre + " hace quikquik..")